bson/src/main/org/bson/codecs/pojo/PojoBuilderHelper.java

```diff
@@ -254,13 +303,33 @@ private static TypeParameterMap getTypeParameterMap(final List<String> genericTy
         return builder.build();
     }
 
+    private static <T> List<String> getGenericTypeNames(final Class<T> clazz) {
+        List<String> genericTypeNames = new ArrayList<String>();
+        for (TypeVariable<? extends Class<? super T>> classTypeVariable : clazz.getTypeParameters()) {
+            genericTypeNames.add(classTypeVariable.getName());
+        }
+        return genericTypeNames;
+    }
+
     static <V> V stateNotNull(final String property, final V value) {
         if (value == null) {
             throw new IllegalStateException(format("%s cannot be null", property));
         }
         return value;
     }
 
+    static boolean isRecord(final Class<?> clazz) {
+        try {
+            clazz.getClass().getMethod("isRecord");
```

Comment:
Do you mean to get the Class of the Class<?> here ?

You could just: `return (Boolean) clazz.getMethod("isRecord").invoke()`

Reply:
No this is actually intentional. isRecord is a method on the class, not the object, so to see if it exists I needed to get the class of the class. I added a comment in the code attempting to explain this. I changed the format to how you did it with invoke though because it's clearer.

Reply:
Cool, thanks for the clarification!